src/zenml/post_execution/artifact.py

```diff
@@ -113,3 +113,10 @@ def __repr__(self) -> str:
             f"type='{self._type}', uri='{self._uri}', "
             f"materializer='{self._materializer}')"
         )
+
+    def __eq__(self, other: Any) -> bool:
+        """Returns whether the other object is referring to the
+        same artifact."""
+        if isinstance(other, ArtifactView):
+            return self._id == other._id and self._uri == other._uri
+        return NotImplemented
```

Comment:
Is it normal/standard to return `NotImplemented` in this way?

Reply:
Yes I think so, if I remember correctly returning `NotImplemented` in `__eq__` is the best practice if you can't do the comparison yourself.

For example if someone decides to subclass ArtifactView (and doesn't do any major changes, let's say just add a small method to it), an object of that class might still be 'equal' to an ArtifactView object. But we can't check for that here in `__eq__` as we don't know of the subclass.
If we return `NotImplemented`, python will as a next step try calling `__eq__` on the other object (the subclass in our example, which might handle comparing to ArtifactView) and if that returns `NotImplemented` as well it will fallback to comparing if it's the same object in memory.

Reply:
Ok thank you. I didn't know that.